bibtex-actions.el

```diff
@@ -169,6 +169,32 @@ and nil means no action."
   :group 'bibtex-actions
   :type 'function)
 
+(defcustom bibtex-actions-major-mode-functions
+  '(((latex-mode) .
+     ((local-bib-files . bibtex-actions-latex--local-bib-files)
+      (keys-at-point . bibtex-actions-latex--keys-at-point)))
+    ((org-mode) .
+     ((local-bib-files . org-cite-list-bibliography-files)
+      (keys-at-point . bibtex-actions-get-key-org-cite))))
+  "The variable determining the major mode specifc functionality.
+It is alist with keys being a list of major modes. The value is an alist
+with values being functions to be used for these modes while the keys
+are symbols used to lookup them up. The keys are
+
+local-bib-files: the corresponding functions should return the list of
+local bibliography files.
+
+insert-keys: the corresponding function should insert the list of keys given
+to as the argument at point in the buffer.
+
+keys-at-point: the corresponding function should return the list of keys at
+point."
+  :group 'bibtex-actions
+  :type '(alist :key-type (repeat string :tag "Major modes")
+                :value-type (set (cons (const local-bib-files) function)
+                                 (cons (const insert-keys) function)
+                                 (cons (const keys-at-pont function)))))
```

Comment:
The idea here is to gather the functionality needed to support a major mode. Are there more places we need mode specific functionality? I think `insert-citation` is one.